pnet_packet/Cargo.toml

```diff
@@ -0,0 +1,29 @@
+[package]
+name = "pnet_packet"
+version = "0.19.0"
+authors = ["Robert Clipsham <robert@octarineparrot.com>"]
+license = "MIT/Apache-2.0"
+homepage = "https://github.com/libpnet/libpnet"
+repository = "https://github.com/libpnet/libpnet"
+description = "Cross-platform, binary packet parsing and manipulation"
+keywords = ["networking", "packet", "parsing", "raw"]
+categories = ["network-programming", "parser-implementations"]
+build = "build.rs"
+
+[features]
+default = ["with-syntex"]
+nightly = ["pnet_macros_plugin"]
+with-syntex = ["syntex", "pnet_macros/with-syntex"]
+
+[dependencies]
+pnet_base = { path = "../pnet_base" }
+pnet_macros = { path = "../pnet_macros", optional = true, version = ">=0.9" }
+pnet_macros_support = { path = "../pnet_macros_support", version = ">=0.1" }
+pnet_macros_plugin = { path = "../pnet_macros_plugin", optional = true, version = ">=0.1" }
+
+syntex = { version = "0.42", optional = true }
+
+[build-dependencies]
+glob = "0.2"
+syntex = { version = "0.42", optional = true }
+pnet_macros = { path = "../pnet_macros", version = ">=0.13" }
```

Comment:
How come the `build-dependencies` and the normal `dependencies` depend on different versions of `pnet_macros`? Is it for a real reason, or just because they were added at different times?

I never specify version requirements on dependencies given by a local `path` at all. I have personally never had the need. But there might be one here that I don't know about. If not, it would be cleaner to just get rid of them. Since they are in the same repo one would always use the one that was the newest at any given checkout anyway.

Reply:
I think the versions might be needed when you publish on crates.io? If not I'll remove them. They're completely arbitrary and probably wrong - it's likely you need the latest versions of everything for it to work.

I have two theories for the different versions:
 * I just blindly copy/pasted from the old Cargo.toml's, and they probably just fell out of sync
 * There (was, not sure if there still is) a weird circular dependency between `pnet` and `pnet_macros` crates, where `pnet` requires `pnet_macros` to build, and `pnet_macros` requires `pnet` to run tests. I'll investigate this and clean it up if possible.

Reply:
You can't publish `path` dependencies to crates.io no. Maybe you can if you have the version specified as well? I have no idea, if so then *mind blown*.

Reply:
When having multiple crates in the same repo like this and I publish them I temporarily change the dependency from being a `path` to being a published version (without a `path`) temporarily and publish, then just revert that from my working directory.

Reply:
According to http://doc.crates.io/specifying-dependencies.html#specifying-path-dependencies you can have both path and version - version is used on crates.io, and path is used locally.

Reply:
Cool. If so I guess we should keep the versions, but update them to the latest ones.